.circleci/config.yml

```diff
@@ -96,6 +96,23 @@ jobs:
             - ./source
             - ./conv.xsl
 
+  # Check circular dependencies with cpp-dependencies tool
+  check_circular_deps:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /hpx
+      - run:
+          name: Check circular dependencies with cpp-dependencies
+          command: |
+              cpp-dependencies --dir /hpx/source/libs --graph-cycles /hpx/tools/cpp-dependencies/circular_deps.dot
+              dot /hpx/tools/cpp-dependencies/circular_deps.dot -Tsvg -o /hpx/tools/cpp-dependencies/circular_deps.svg
+              if [[ $(wc -l /hpx/tools/cpp-dependencies/circular_deps.dot | awk '{print $1}') -gt 2 ]]; then exit 1; fi
+      - store_artifacts:
+          path:
+            - /hpx/tools/cpp-dependencies/circular_deps.dot
+            - /hpx/tools/cpp-dependencies/circular_deps.svg
```

Comment:
I think these have to go in two separate `- store_artifacts` entries (see https://circleci.com/gh/STEllAR-GROUP/hpx/116132 and https://circleci.com/docs/2.0/configuration-reference/#store_artifacts).